### ISSUE TITLE
Creating Quantity with Quantity

### ISSUE THREAD

**Issue by jules-ch** [hgrecco/pint#1231]
When creating a new quantity, I encountered some weird behavior, where you could create a Quantity with a Quantity as magnitude.

This behaviour feels weird, I don't know if anyone used this but well it's possible with the current API.

Example : 

```python

>>> from pint import UnitRegistry
>>> ureg = UnitRegistry
>>> air_density = 1.225 * ureg("kg/m**3")
>>> ureg.Quantity(air_density)
<Quantity(1.225, 'kilogram / meter ** 3')>
>>> ureg.Quantity(air_density, "kg/m**3")
<Quantity(1.225 kilogram / meter ** 3, 'kilogram / meter ** 3')>
>>> ureg.Quantity(air_density, "kg/m**3").m
<Quantity(1.225, 'kilogram / meter ** 3')>
```

I would have expected to throw an error.

We might have better options:

- Try to convert to the units specified.
It can be used when a function accept both Quantity or base type of magnitude.
I'm using a lot of those functions and I end up with something like this:

```python
def __init__(air_density: Union[float, Quantity]):
  if isinstance(air_density, Quantity):
      self.air_density = air_density.to("kg/m**3")
  else:
      self.air_density = units.Quantity(air_density, "kg/m**3")
```

- Multiply Quantity with units
This one feels weird though.

what do you think ? safest choice is throw an error

**Comment by hgrecco** [comment]
While this does not constitute an answer, I would like to point out that a related question has been asked in the past. What is the right behavior for:
```
q2 = Quantity(m, q1)
```
where `m` is a number and `q1` is a quantity object. This construct has appeared within functions that accept units or quantities and also within the discussion if units are equivalent to quantities with unit magnitude. I think we need to look at that as well:

https://github.com/hgrecco/pint/blob/012bb5c8ba12a90c0746a97ea55d289fef8387ac/pint/quantity.py#L233

**Comment by jules-ch** [comment]
Good point I encountered this use case aswell where units were defined with a scale which was not a prefix.

Overall we should clearly state how we can use Quantity with itself (or not).
